proposal/sgaumi/proposal.rst

```diff
@@ -0,0 +1,70 @@
+===========================================
+Proposal for Compression/Decompression Tool
+===========================================
+
+
+Basic information
+=================
+
+GitHub repository : `https://github.com/sgaumi/compression_decompression_tool <https://github.com/sgaumi/compression_decompression_tool>`__ 
+
+The purpose of this project is to make a software to compress and decompress files.
+
+Problem to solve
+================
+
+It’s sometimes interesting or even necessary to compress files to free some places on memory computer (or any storage device) or to send these files through a network. It’s also necessary to be able to decompress these files to get back the information.
+To perform file compression it’s necessary to encode the data in a different way so that it takes less places.
+To solve this problem the software will only use a Huffman coding in a first version.
+In a second version (If the first version is not too long to implement) the software will use both a Huffman coding and LZ77. The combination of these two algorithm is known as deflate which is the algorithm used by gzip (the GNU compression software).
+
+Perspective users
+=================
+
+In a first version, the user will be able to use the software through a Bash Shell with typing the software command and giving the file name (the path) and the options, which allow the user to choose the compression or decompression mode, the algorithm chosen (Huffman static, Huffman adaptive or deflate in a second version).
+In a second version, the user will be able to use software by executing an application and give the file name (path) and options through a graphic user interface. 
+
+System architecture
+===================
+
+For compression, the system takes the path of a file as input and opens the file in reading mode. It creates on other file where the compression will be performed. At the end the system returns the file created and close the original file (it’s not deleted). 
+For decompression, the system takes the path of a compressed file (it will work only with a file compressed by this software, it is not designed to support files compressed by other software) as input and opens the file in reading mode. It creates on other file where the decompression will be performed. At the end the system returns the file created and close the original compressed file (it’s not deleted). 
+
+API description
+===============
+
+I have no idea what I will provide for the API for the moment, it will depend how I choose to implement the system. 
+Maybe for the Huffman tree in C++, it would be created as an object with some features and would be used by a user. 
+Maybe for the compression / decompression in python, it would be possible to override the method chosen for compression.
```

Comment:
This is a good way to pursue.  If you make a really good codec, it will be worth to discuss with the CPython community whether or not they should include it to the official release.

By making an optional codec, you can also benchmark against those already built in.